lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -78,8 +109,60 @@ public function __construct($directory, $extension = '')
             ));
         }
 
+        $extension = (string) $extension;
         $this->directory = realpath($directory);
-        $this->extension = (string) $extension;
+        $this->extension = $extension ?: $this->extension;
+        $this->fileMode = 0666 & ~umask();
+    }
+
+    /**
+     * Sets the mode that new directories will be created with.
+     * @param int $mode Mode, normally in octal (ex: 0755)
+     */
+    public function setDirectoryMode($mode)
```

Comment:
My main issue is the umask is a convoluted mess so people will not use it, and in any case, the default file mode of 0777 is even worse.  At a minimum, perhaps the default dir perms should be 0775 and the default file perms 0664 so the cache files are not world-executable.  Are you suggesting we have a umask setter on this class, or that the user should change his umask before using this class?  Changing the global umask hardly seems like a good idea, especially in a multithreaded environment (`umask()` is not thread-safe).  This method would also make it impossible to have two cache instances with different permissions.  Setting the umask for this cache adapter itself (like my proposed mode setters) makes some sense, although I really don't see the point of having the user set the umask when they don't even know the base mode.  Furthermore, my issue with the cache files being executable remains unless you did something like `$dirUmask = 0022; $fileUmask = 0133;`.